client/src/App.tsx

```diff
@@ -69,6 +72,35 @@ const onClickP1 = (
   return true;
 }
 
+/**
+ * The Sign-In client object.
+ */
+let auth2: any;
+let googleUser: any;
+
+export const googleAPILoaded: Promise<void> = new Promise(resolve => {
+  gapi.load('auth2', () => {
+    /**
+     * Retrieve the singleton for the GoogleAuth library and set up the
+     * client.
+     */
+    gapi.auth2.init({
+        client_id: '634069824484-ch6gklc2fevg9852aohe6sv2ctq7icbk.apps.googleusercontent.com'
+    }).then( function() {
+        // Sign in the user if they are currently signed in.
+        auth2 = gapi.auth2.getAuthInstance(); 
+        if (auth2.isSignedIn.get() == true) {
+          const googleUser = gapi.auth2.getAuthInstance().currentUser.get();
+          const profile = googleUser.getBasicProfile();
+          store.dispatch(loginSuccess(profile.getName(), profile.getName(), profile.getEmail())); 
+        }
+      }
+    );
+
+    resolve();
+  });
+});
+
```

Comment:
This logic shouldn't go in `App.tsx`. Only really high level code should go here (like the routing container). Can we move this into a new action in the `login/actions.ts` file? Then we'll call that action in the `componentDidMount` function the header component.

Reply:
Well the UI will display as soon as the page is loaded. We could do the initialization from the `componentDidMount` function in `App.tsx`, but we should do it by calling an action, not by putting all the code for it there